### PR TITLE
Inversion Vector of a Singly Linked List

### LINKED ISSUE #4115: Inversion Vector of a Singly Linked List
## 🚀 Feature
Inversion vector for a given permutation in CPP language.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yep!

## Problem
For a singly linked list X of integers, a pair (xi,xj) is called an inversion of X if for i < j , xi > xj. The inversion vector of X is the sequence of integers d1, d2,.. dn such that dj is the number of elements greater than xj and to its left in the sequence, so 0 <= dj < j. Compute the inversion vector for a given permutation.

### PR DESCRIPTION
**Pull request by Suvraneel**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yep !

### Description
Inversion Vector of a Singly Linked List

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

resolves #4115

### CODE REVIEW

**Comment by Suvraneel** [comment]
Hello maintainers!
Kindly review this... Followed all suggestions from previous PRs :)
@HarshCasper @bhav09  @siddharth25pandey

File: C-Plus-Plus/ds/inversion_vector.cpp [hunk 1]
```diff
@@ -0,0 +1,209 @@
+/****************************************************************************
+Author - @Suvraneel 
+        Suvraneel Bhuin
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/inversion_vector.cpp]: Remove this redundant comments

File: C-Plus-Plus/ds/inversion_vector.cpp [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+{
+    int value;
+    struct node *next;
+}node;
+
+//function list
+node* create_node(int);
+void insert_node();
+void display();
+void inversion();
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/inversion_vector.cpp]: Remove the function declarations

File: C-Plus-Plus/ds/inversion_vector.cpp [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+            cout << "Program was terminated";
+            exit(0);
+
+        default:
+            cout << endl << "Invalid Choice";
+            break;
+        }
+    }
+    return 0;
+ }
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/inversion_vector.cpp]: Shift the main after all the functions

File: C-Plus-Plus/ds/inversion_vector.cpp [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+{
+    if (first == NULL)
+        cout << endl << "No nodes in the list";
+    else
+        for (ptr = first;ptr != NULL;ptr = ptr->next)
+            cout << ptr->value << " -> ";
+}
+
+/*
+Time Complexity of Inversion Vector = O(n^2)
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/inversion_vector.cpp:142]: Shift TC after the end of outputs and write SC as well

File: C-Plus-Plus/ds/inversion_vector.cpp [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+Enter value of the Node:        8
+8 inserted as node
+
+Enter your choice : 1
+Enter value of the Node:        6
+6 inserted as node
+
+Enter your choice : 1
+Enter value of the Node:        9
+9 inserted as node
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/inversion_vector.cpp]: Remove the extra elements show with 5 elements

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Suvraneel** [comment]
@ankitaggarwal23  Done :)

**Review by ankitaggarwal23 — APPROVED** [review]

**Comment by HarshCasper** [comment]
Please make an issue here and raise a PR, for an article as part of our Docs repository: https://github.com/TesseractCoding/NeoAlgo-Docs

It would be much appreciated. Thanks!

**Pull request merged**